master/buildbot/steps/trigger.py

```diff
@@ -167,14 +189,18 @@ def getAllGotRevisions(self):
         return all_got_revisions
 
     @defer.inlineCallbacks
-    def worstStatus(self, overall_results, rclist):
+    def worstStatus(self, overall_results, rclist, unimportant_brids):
         for was_cb, results in rclist:
             if isinstance(results, tuple):
-                results, _ = results
+                results, brids_dict = results
 
             if not was_cb:
                 yield self.addLogWithFailure(results)
                 results = EXCEPTION
+
+            # continue if this brid has to be ignored
+            if len(set(unimportant_brids) & set(brids_dict.values())) != 0:
```

Comment:
It is indeed a complexity of the difference between buildsets and buildrequests.

In this particular case schedulers are marked unimportants. There is one bsid by scheduler, and then theorically one brid by builder (in the case the scheduler triggers several builders).
For the usecase we are working on this is highly unlikely, but we should take care of this corner case though.

So I think this line is correct (it should use superset though), but it might deserve some explaination in comment